### PR TITLE
#167765407 User can make a comment

### PR DESCRIPTION
**Pull request by okobaba1**
#### What does this PR do?
It allows a user to post a comment
#### How should this be manually tested?
npm run test
#### Any background context you want to provide?
run `npm run reset-db`
#### What are the relevant pivotal tracker stories?
[#167765407](https://www.pivotaltracker.com/story/show/167765407)
#### Implementation Plan
https://docs.google.com/document/d/1U-7NJAkENDf5FgZWw6eWUQt3WSp1U6P8-_3VMzlpEkU/edit#

### CODE REVIEW

File: src/services/commentServices.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import models from '../models';
+
+/**
+ * Comment service Class
+ */
+export default class commentServices {
+  /**
+   * @name CreateComment
+   * @description Interacts with model to create comment
+   * @param { string } userData the user's data
```
Comment (victor-nach) [inline src/services/commentServices.js:10]: I think it's better you use comment here instead of userData

Comment (okobaba1) [reply src/services/commentServices.js:10]: I used that because I'm also sending in some other details too, userId inclusive

Comment (debelistic) [reply src/services/commentServices.js:10]: Nice setup

**Review by debelistic — APPROVED** [review]

File: public/api-docs/swaggerDoc.json [hunk 2]
```diff
@@ -314,6 +314,45 @@
...
+                                }
+                            }
+                        }
+                    }
+                ],
+                "responses": {
+                    "201": {
+                        "description": "successful operation",
+                        "schema": {
+                            "$ref": "#/definitions/User"
```
Comment (voke1) [inline public/api-docs/swaggerDoc.json]: you may want to create a new response definition as I don't think this `User defintions` tally with the response the api returns

File: src/controllers/commentController.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const { resSuccess, resError } = ResponseMsg;
+
+
+/**
+ * Comments controller Class
+ */
+export default class CommentController {
+  /**
+     * @name postComment
+     * @description Allows a user make a one way trip
```
Comment (voke1) [inline src/controllers/commentController.js]: Sorry, does postComment makes me make a one way trip? Please, check this

File: src/database/migrations/20190914142540-create-comment.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    userId: {
+      type: Sequelize.INTEGER,
+      allowNull: false,
+    },
+    comment: {
+      type: Sequelize.TEXT,
+      allowNull: false,
+    },
+    isVisible: {
```
Comment (voke1) [inline src/database/migrations/20190914142540-create-comment.js:21]: Please, why do we have this `isVisible` field?

File: src/models/comment.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    isVisible: {
+      type: DataTypes.BOOLEAN,
+      allowNull: false,
+      defaultValue: true,
+    }
+  }, {});
+  Comment.associate = (models) => {
+    Comment.belongsTo(models.Trip, {
+      foreignKey: 'tripId', as: 'trips', timestamps: false
+    });
```
Comment (voke1) [inline src/models/comment.js:24]: I believe there should be a relationship between Comment  and User and this should be defined by an associate function

File: src/database/migrations/20190914142540-create-comment.js (hunk 4 above)
Comment (okobaba1) [reply src/database/migrations/20190914142540-create-comment.js:21]: An admin might want to hide a comment

File: src/models/comment.js (hunk 5 above)
Comment (okobaba1) [reply src/models/comment.js:24]: done

File: src/controllers/commentController.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+     */
+  static async postComment(req, res) {
+    try {
+      const { id: userId } = req.user;
+      const { comment } = req.body;
+      const { tripId } = req.params;
+      const data = await commentServices.createComment({ userId, comment, tripId });
+      return resSuccess(res, 201, data);
+    } catch (error) {
+      console.log(error.message);
```
Comment (anneKay) [inline src/controllers/commentController.js]: Remove console statement

File: src/controllers/commentController.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      return resError(res, 500, error.message);
+    }
+  }
+}
```
Comment (anneKay) [inline src/controllers/commentController.js]: Add new line

File: src/models/comment.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
```
Comment (anneKay) [inline src/models/comment.js]: Add an `onDelete: 'CASCADE'` here

File: src/models/comment.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    Comment.belongsTo(models.User, {
+      foreignKey: 'userId', as: 'users', timestamps: false
+    });
```
Comment (anneKay) [inline src/models/comment.js]: Do same here

File: src/models/trips.js [hunk 10]
```diff
@@ -29,7 +29,10 @@ export default (sequelize, DataTypes) => {
         key: 'id'
       },
       allowNull: false
-    },
-  });
+    }
+  }, {});
+  Trip.associate = (models) => {
+    Trip.hasMany(models.Comment, { foreignKey: 'tripId' });
```
Comment (anneKay) [inline src/models/trips.js]: Add an ` onDelete: 'CASCADE',` option here

File: src/validation/commentValidation.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import { check } from 'express-validator';
+
+const commentInputValidation = [
+  check('comment').trim().not().isEmpty()
+    .withMessage('Origin field cannot be empty')
```
Comment (anneKay) [inline src/validation/commentValidation.js]: Please use a more descriptive message here

**Pull request merged**